// This source file is part of the Swift.org open source project
//
// Copyright (c) 2024-2025 Apple Inc. and the Swift project authors
// Licensed under Apache License v2.0 with Runtime Library Exception
//
// See http://swift.org/LICENSE.txt for license information
// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors

syntax = "proto3";

package llbuild3;

import "llbuild3/CAS.proto";
import "llbuild3/Label.proto";

// Enum representing the supported action cache key types
enum CacheKeyType {
  CACHE_KEY_TYPE_TASK = 0;
  CACHE_KEY_TYPE_ACTION = 1;
}

message CacheKey {
  Label label = 1;
  
  CacheKeyType type = 2;
  
  CASObjectID content = 3;
}

message CacheValue {
  CASObjectID data = 1;
  
  repeated ActionStat stats = 2;
}

message ActionStat {
  string name = 1;
  oneof value {
    int64 intValue = 2;
    string stringValue = 3;
    bool boolValue = 4;
    double doubleValue = 5;
    CASObjectID casObject = 6;
  }
}
